src/panels/lovelace/common/evaluate-filter.ts

```diff
@@ -1,12 +1,31 @@
 import { HassEntity } from "home-assistant-js-websocket";
+import { HomeAssistant } from "../../../types";
+
+export const evaluateFilter = (
+  hass: HomeAssistant,
+  current_entity_id: string,
+  filter: any
+): boolean => {
+  const stateObj: HassEntity = hass.states[current_entity_id];
+
+  if (!stateObj) {
+    return false;
+  }
 
-export const evaluateFilter = (stateObj: HassEntity, filter: any): boolean => {
   const operator = filter.operator || "==";
   let value = filter.value ?? filter;
   let state = filter.attribute
     ? stateObj.attributes[filter.attribute]
     : stateObj.state;
 
+  if (typeof value === "string" && value.includes(".")) {
+    value = hass.states[value]?.state;
+
+    if (!value) {
```

Comment:
```suggestion
    if (value === undefined) {
```

I believe an empty string may be a legitimate state value, in which case we can allow that to pass through for comparison.

Reply:
And for 0